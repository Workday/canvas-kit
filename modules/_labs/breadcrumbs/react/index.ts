import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
