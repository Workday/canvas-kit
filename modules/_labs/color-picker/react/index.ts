export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
