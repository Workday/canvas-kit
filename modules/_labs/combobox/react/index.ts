import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
