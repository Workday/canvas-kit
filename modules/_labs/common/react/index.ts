export * from './lib/Flex';
export * from './lib/Box';
