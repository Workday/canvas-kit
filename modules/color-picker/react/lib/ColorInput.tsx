import * as React from 'react';
import {styled, Themeable, ContentDirection} from '@workday/canvas-kit-labs-react-core';
import {expandHex, GrowthBehavior, ErrorType} from '@workday/canvas-kit-react-common';
import {colors, spacing, type, inputColors} from '@workday/canvas-kit-react-core';
import TextInput, {TextInputProps} from '@workday/canvas-kit-react-text-input';
import {ColorSwatch} from './parts/ColorSwatch';

export interface ColorInputProps extends Themeable, TextInputProps, GrowthBehavior {
  /**
   * The value of the ColorInput.
   * @default ''
   */
  value?: string;
  /**
   * If true, show a checkmark in the swatch tile when a custom hex color is entered in the ColorInput.
   * @default false
   */
  showCheck?: boolean;
  /**
   * The placeholder text of the ColorInput.
   * @default FFFFFF
   */
  placeholder?: string;
  /**
   * The type of error associated with the ColorInput (if applicable).
   */
  error?: ErrorType;
  /**
   * The function called when the ColorInput state changes.
   */
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  /**
   * The function called when a valid hex value is entered in the ColorInput. The `color` argument passed to the callback function is prefixed with a hash and expanded if necessary (e.g., `03F` is converted to `#0033FF`).
   */
  onValidColorChange?: (color: string) => void;
}

const colorInputWidth = 116;

const CustomHexInput = styled(TextInput)<Pick<ColorInputProps, 'disabled' | 'grow'>>(
  {
    boxSizing: 'border-box',
    minWidth: colorInputWidth,
    width: colorInputWidth,
    ...type.variant.mono,
    '&:focus::placeholder': {
      color: 'transparent',
    },
  },
  ({grow}) =>
    grow && {
      minWidth: '100%',
      width: '100%',
    },
  ({disabled}) => ({
    backgroundColor: disabled ? colors.soap200 : '',
  }),
  ({theme}) => ({
    paddingLeft:
      theme.direction === ContentDirection.LTR ? '46px' : 'calc(100% - 86px) /* @noflip */',
    // We're using @noflip because ColorInput should stay LTR, therefore, we need to adjust the padding using ContentDirection, not using rtl-css-js.
  })
);

const ColorInputContainer = styled('div')<Pick<ColorInputProps, 'grow'>>(
  {
    position: 'relative',
    width: colorInputWidth,
  },
  ({grow}) =>
    grow && {
      minWidth: '100%',
      width: '100%',
    }
);

const PoundSignPrefix = styled('span')<Pick<ColorInputProps, 'disabled'>>(
  {
    position: 'absolute',
    top: 10,
    ...type.body,
    ...type.variant.mono,
  },
  ({disabled}) => ({
    color: disabled ? inputColors.disabled.text : undefined,
  }),
  ({theme}) => ({
    left: theme.direction === ContentDirection.LTR ? '36px' : '88px',
    //    - LTR -> left: 36px;
    //    - RTL -> right: 88px;
    //
    // It's because we're changing the value of this attribute based on ContentDirection but the attribute itself is getting flipped by rtl-css-js.
    // We're doing this to adjust the spacing of the # sign as it should always stay on the left of the input in both LTR and RTL.
  })
);

const SwatchTile = styled(ColorSwatch)({
  position: 'absolute',
  top: 0,
  left: spacing.xxs,
  marginTop: '10px', // Fix vertical alignment on IE11
  boxShadow: 'inset 0 0 0 1px rgba(0,0,0,0.25)',
  pointerEvents: 'none',
});

export default class ColorInput extends React.Component<ColorInputProps> {
  public render() {
    // TODO: Standardize on prop spread location (see #150)
    const {
      placeholder = 'FFFFFF',
      value = '',
      showCheck,
      onChange,
      onValidColorChange,
      inputRef,
      disabled,
      error,
      grow,
      ...elemProps
    } = this.props;
    const formattedValue = this.formatValue(value);

    return (
      <ColorInputContainer grow={grow}>
        <CustomHexInput
          dir="ltr"
          inputRef={inputRef}
          onChange={this.handleChange}
          type="text"
          placeholder={placeholder}
          value={formattedValue}
          error={error}
          spellCheck={false}
          disabled={disabled}
          grow={grow}
          maxLength={7} // 7 to allow pasting with a hash
          {...elemProps}
        />
        <SwatchTile
          showCheck={showCheck}
          color={this.isValidHex(formattedValue) ? `#${formattedValue}` : ''}
        />
        <PoundSignPrefix disabled={disabled}>#</PoundSignPrefix>
      </ColorInputContainer>
    );
  }

  private handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    const value = this.formatValue(e.currentTarget.value);

    if (this.props.onChange) {
      this.props.onChange(e);
    }

    if (this.isValidHex(value) && this.props.onValidColorChange) {
      this.props.onValidColorChange(`#${expandHex(value)}`);
    }
  };

  private isValidHex = (value: string) => {
    return /(^#?[0-9A-F]{3}$)|(^#?[0-9A-F]{6}$)/i.test(value);
  };

  private formatValue = (value: string) => {
    return value.replace(/#/g, '').substring(0, 6);
  };
}
