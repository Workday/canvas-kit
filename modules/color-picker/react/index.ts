import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
export * from './lib/parts/ColorSwatch';
