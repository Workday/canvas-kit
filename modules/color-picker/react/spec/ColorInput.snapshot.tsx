import * as React from 'react';
import ColorInput from '../lib/ColorInput';
import * as renderer from 'react-test-renderer';

describe('ColorInput Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={''} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'e6e'} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with a selected color without #', () => {
    const component = renderer.create(
      <ColorInput onChange={jest.fn()} onValidColorChange={jest.fn()} value={'#e6e'} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a light checkIcon if hex value is dark', () => {
    const component = renderer.create(
      <ColorInput
        showCheck={true}
        onChange={jest.fn()}
        onValidColorChange={jest.fn()}
        value={'#e6e'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a dark checkIcon if hex value is light', () => {
    const component = renderer.create(
      <ColorInput
        showCheck={true}
        onChange={jest.fn()}
        onValidColorChange={jest.fn()}
        value={'#ffffff'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a disabled input', () => {
    const component = renderer.create(
      <ColorInput
        disabled={true}
        onChange={jest.fn()}
        onValidColorChange={jest.fn()}
        value={'#ffffff'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders valid empty color input', () => {
    const component = renderer.create(
      <ColorInput disabled={false} onChange={jest.fn()} onValidColorChange={jest.fn()} value={''} />
    );
    expect(component).toMatchSnapshot();
  });
});
