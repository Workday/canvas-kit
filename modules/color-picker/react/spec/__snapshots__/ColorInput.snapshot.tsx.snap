// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput Snapshots renders a dark checkIcon if hex value is light 1`] = `
.emotion-8 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-6 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-5 {
  display: inline-block;
  position: absolute;
  left: 6px;
  top: 8px;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #000000;
}

.emotion-5:hover .wd-icon-fill {
  fill: #000000;
}

.emotion-5 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-5:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value="ffffff"
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  />
  <span
    className="emotion-5"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
  />
  <span
    className="emotion-6 emotion-7"
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders a disabled input 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: #f0f1f2;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-5 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  color: #a1aab3;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      disabled={true}
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value="ffffff"
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  />
  <span
    className="emotion-5 emotion-6"
    disabled={true}
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders a light checkIcon if hex value is dark 1`] = `
.emotion-8 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-6 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-5 {
  display: inline-block;
  position: absolute;
  left: 6px;
  top: 8px;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-5:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-5 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-5:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value="e6e"
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "#e6e",
      }
    }
  />
  <span
    className="emotion-5"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
      }
    }
  />
  <span
    className="emotion-6 emotion-7"
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders as expected 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-5 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value=""
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <span
    className="emotion-5 emotion-6"
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders valid empty color input 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-5 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      disabled={false}
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value=""
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <span
    className="emotion-5 emotion-6"
    disabled={false}
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders with a selected color 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-5 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value="e6e"
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "#e6e",
      }
    }
  />
  <span
    className="emotion-5 emotion-6"
  >
    #
  </span>
</div>
`;

exports[`ColorInput Snapshots renders with a selected color without # 1`] = `
.emotion-7 {
  position: relative;
}

.emotion-2 {
  display: inline-block;
  position: relative;
}

.emotion-1 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  box-sizing: border-box;
  padding-left: 46px;
  min-width: 116px;
  width: 116px;
  background-color: px;
}

.emotion-1::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-1::-moz-placeholder {
  color: #5e6a75;
}

.emotion-1:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-1::placeholder {
  color: #5e6a75;
}

.emotion-1:hover {
  border-color: #333d47;
}

.emotion-1:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-1:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-1:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-1:disabled::placeholder {
  color: #a1aab3;
}

.emotion-1:focus::-webkit-input-placeholder {
  color: transparent;
}

.emotion-1:focus::-moz-placeholder {
  color: transparent;
}

.emotion-1:focus:-ms-input-placeholder {
  color: transparent;
}

.emotion-1:focus::placeholder {
  color: transparent;
}

.emotion-3 {
  position: absolute;
  cursor: pointer;
  height: 20px;
  width: 20px;
  top: 0px;
  bottom: 0px;
  left: 8px;
  margin: auto;
  margin-top: 10px;
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.25);
  pointer-events: none;
  border-radius: 2px;
}

.emotion-5 {
  position: absolute;
  left: 36px;
  top: 10px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #5e6a75;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-2"
  >
    <input
      className="emotion-0 emotion-1"
      maxLength={7}
      onChange={[Function]}
      placeholder="FFFFFF"
      spellCheck={false}
      type="text"
      value="e6e"
    />
  </div>
  <div
    className="emotion-3 emotion-4"
    style={
      Object {
        "backgroundColor": "#e6e",
      }
    }
  />
  <span
    className="emotion-5 emotion-6"
  >
    #
  </span>
</div>
`;
